dsf-gdb/org.eclipse.cdt.tests.dsf.gdb/src/org/eclipse/cdt/tests/dsf/gdb/tests/nonstop/MultiProcessRemoteTest.java

```diff
@@ -0,0 +1,242 @@
+		super.doBeforeTest();
+
+		final DsfSession session = getGDBLaunch().getSession();
+
+		Runnable runnable = () -> {
+			fServicesTracker = new DsfServicesTracker(TestsPlugin.getBundleContext(), session.getId());
+			fGdbProcesses = fServicesTracker.getService(IGDBProcesses.class);
+			fCommandControl = fServicesTracker.getService(ICommandControlService.class);
+			fGDBBackend = fServicesTracker.getService(IGDBBackend.class);
+		};
+		session.getExecutor().submit(runnable).get();
+	}
+
+	@Override
+	protected void setLaunchAttributes() {
+		super.setLaunchAttributes();
+
+		setLaunchAttribute(IGDBLaunchConfigurationConstants.ATTR_DEBUGGER_NON_STOP, true);
+		setLaunchAttribute(ICDTLaunchConfigurationConstants.ATTR_PROGRAM_NAME, getBinary());
+		setLaunchAttribute(ICDTLaunchConfigurationConstants.ATTR_DEBUGGER_STOP_AT_MAIN, false);
+		setLaunchAttribute(ICDTLaunchConfigurationConstants.ATTR_DEBUGGER_START_MODE,
+				IGDBLaunchConfigurationConstants.DEBUGGER_MODE_REMOTE_ATTACH);
+		setLaunchAttribute(ITestConstants.LAUNCH_GDB_SERVER_MULTI, Boolean.TRUE.toString());
+		setLaunchAttribute(IGDBLaunchConfigurationConstants.ATTR_DEBUGGER_REMOTE_BINARY,
+				getLaunchAttribute(ICDTLaunchConfigurationConstants.ATTR_PROGRAM_NAME));
+		if (testName.getMethodName().contains("_noProgram"))
+			setLaunchAttribute(ITestConstants.LAUNCH_GDB_SERVER_WITHOUT_PROGRAM, Boolean.TRUE.toString());
+	}
+
+	@Override
+	public void doAfterTest() throws Exception {
+		if (fServicesTracker != null)
+			fServicesTracker.dispose();
+		super.doAfterTest();
+		for (Process p : appProcesses) {
+			if (p.isAlive())
+				p.destroyForcibly();
+		}
+		appProcesses.clear();
+	}
+
+	@Override
+	protected boolean isTargetExpectedToStopAfterLaunch() {
+		// For some tests, we'll get stop event but not for others. Checking stop event
+		// at start of launch is not required for this test class
+		return false;
+	}
+
+	@Test
+	public void launchGdbServerWithProgramAndStartMulipleNewExecutables() throws Throwable {
+		assertEquals("Wrong number of containers at the start of launch", SyncUtil.getAllContainerContexts().length,
+				getInitialProcessCount());
+
+		final IPath execPath = fGDBBackend.getProgramPath();
+		Map<String, Object> attributes = getLaunchConfiguration().getAttributes();
+
+		IMIExecutionDMContext[] ctx = debugNewProcess(fGDBBackend.getProgramPath(), attributes);
+		assertTrue("Process-1 is not started properly as no threads found for it", ctx != null && ctx.length > 0);
+
+		ctx = debugNewProcess(execPath, attributes);
+		assertTrue("Process-2 is not started properly as no threads found for it", ctx != null && ctx.length > 0);
+
+		ctx = debugNewProcess(execPath, attributes);
+		assertTrue("Process-3 is not started properly as no threads found for it", ctx != null && ctx.length > 0);
+	}
+
+	@Test
+	public void launchGdbServerWithoutProgramAndStartMulipleNewExecutables_noProgram() throws Throwable {
+		launchGdbServerWithProgramAndStartMulipleNewExecutables();
+	}
```

Comment:
Was this test supposed to do more?

Reply:
I think the answer to this question is here:

https://github.com/eclipse-cdt/cdt/pull/336/files#diff-4cfbb32f09206759612425a8f3253d379d272eeb8e2e9165d0cb3d355e3a4437R86 (line 86 in this file)

Reply:
Thanks @Torbjorn-Svensson for explaining. I wish we didn't need that, but resolving this now.